Realm/RLMQueryUtil.mm

```diff
@@ -71,6 +71,14 @@ static void RLMPrecondition(bool condition, NSString *name, NSString *format, ..
     return prop;
 }
 
+NSPredicate *RLMValidatedPredicate(id predicateFormat, va_list args) {
+    if (predicateFormat && ![predicateFormat isKindOfClass:[NSString class]]) {
```

Comment:
Again, if memory serves me right I think this was left over from very early code where you could pass in `nil` as a filter, which was the equivalent of `allObjects`. Not useful anymore obviously.